src/js/background/context-menu.js

```diff
@@ -86,65 +113,87 @@ const relayContextMenus = {
 
     // Generate aliases menu item
     // If a user is maxed out/not premium, the generate item will be disabled.
-    const canUserGenerateAliases = await relayContextMenus.utils.getUserStatus.canGenerateMoreAliases();
-    const menuData = canUserGenerateAliases ? staticMenuData.generateAliasEnabled : staticMenuData.generateAliasDisabled;
+    const canUserGenerateAliases =
+      await relayContextMenus.utils.getUserStatus.canGenerateMoreAliases();
+    const menuData = canUserGenerateAliases
+      ? staticMenuData.generateAliasEnabled
+      : staticMenuData.generateAliasDisabled;
     relayContextMenus.menus.create(menuData);
 
-
-    // COMPATIBILITY NOTE: Chrome uses the contextMenus API to create menus. Firefox built their own API, menus, based on it. It has additional features that are only available in Firefox. Anything wrapped in a (browser.menus) check is only executed in a browser that supports it. 
+    // COMPATIBILITY NOTE: Chrome uses the contextMenus API to create menus. Firefox built their own API, menus, based on it. It has additional features that are only available in Firefox. Anything wrapped in a (browser.menus) check is only executed in a browser that supports it.
     if (browser.menus) {
-      const userHasSomeAliasesCreated = (await relayContextMenus.utils.getUserStatus.getNumberOfAliases() > 0);
-      
+      const userHasSomeAliasesCreated =
+        (await relayContextMenus.utils.getUserStatus.getNumberOfAliases()) > 0;
+
       const aliases = await relayContextMenus.utils.getAliases();
 
-      // Create Use Existing Alias submenu
-      if (currentWebsite &&  await relayContextMenus.utils.getGeneratedForHistory(currentWebsite) && userHasSomeAliasesCreated ) {
-        await relayContextMenus.menus.create(staticMenuData.existingAlias, {
-          createExistingAliases: true,
-          parentMenu: staticMenuData.useExistingAliasFromWebsite,
-          exisitingSite: true,
+      const masksWereGeneratedOrUsedOnCurrentWebsite =
+        await relayContextMenus.utils.checkIfAnyMasksWereGeneratedOrUsedOnCurrentWebsite(
           currentWebsite
-        }, aliases);
-      } 
+        );
 
-      // Create "Recent Aliases…" menu
-      if ( userHasSomeAliasesCreated ) {
-        await relayContextMenus.menus.create(staticMenuData.existingAlias, {
-          createExistingAliases: true,
-          parentMenu: staticMenuData.useExistingAlias,
-          exisitingSite: false,
-          currentWebsite
-        }, aliases)
+      // Create Use Existing Alias submenu
+      if (
+        currentWebsite &&
+        masksWereGeneratedOrUsedOnCurrentWebsite &&
+        userHasSomeAliasesCreated
```

Comment:
praise: thank you for creating human-readable variables! Makes it so much easier to understand what is going on.